app/src/main/java/subins2000/manglish/MainActivity.java

```diff
@@ -1,75 +1,96 @@
 /**
  * Manglish, malayalam to manglish converter
  * Copyright (C) 2019 Subin Siby
- *
+ * <p>
  * This program is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
  * the Free Software Foundation, version 3 of the License
- *
+ * <p>
  * This program is distributed in the hope that it will be useful,
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU General Public License for more details.
- *
+ * <p>
  * You should have received a copy of the GNU General Public License
  * along with this program. If not, see <https://www.gnu.org/licenses/>.
  */
 
 package subins2000.manglish;
 
+import android.content.SharedPreferences;
 import android.net.Uri;
 import android.os.Bundle;
+
 import com.google.android.material.navigation.NavigationView;
 
+import android.preference.PreferenceManager;
 import android.view.Menu;
 import android.view.MenuItem;
+import android.view.View;
+import android.view.inputmethod.InputMethodManager;
 
+import androidx.annotation.NonNull;
 import androidx.appcompat.app.ActionBarDrawerToggle;
 import androidx.appcompat.app.AppCompatActivity;
 import androidx.appcompat.widget.Toolbar;
 import androidx.core.view.GravityCompat;
 import androidx.drawerlayout.widget.DrawerLayout;
 import androidx.fragment.app.Fragment;
-import androidx.fragment.app.FragmentTransaction;
 
 public class MainActivity extends AppCompatActivity
-        implements NavigationView.OnNavigationItemSelectedListener, HomeFragment.OnFragmentInteractionListener, AboutFragment.OnFragmentInteractionListener {
+        implements NavigationView.OnNavigationItemSelectedListener,
+        HomeFragment.OnFragmentInteractionListener, AboutFragment.OnFragmentInteractionListener {
+
+    private static final String NAV_ID = "NavId";
 
+    DrawerLayout drawer;
     NavigationView navigationView;
 
-    Fragment homeFragment = new HomeFragment();
```

Comment:
The object was created beforehand and stored to avoid making the object again and again on navigation. Please keep this.

Reply:
You can't avoid recreating fragment by this way when activity recreation happens ;) With this approach, you get overlaid fragments after the screen rotation or another configuration changes.

Reply:
![9Jd-Z21KuUM](https://user-images.githubusercontent.com/37016581/60437044-1a8a5e00-9c16-11e9-9330-1185e5198902.jpg)

Reply:
That I didn't know about. I'm a noob in Android development. Thanks for pointing it out :)

Reply:
I am glad to help :)